### DISCUSSION TITLE
connecting the bolt to  ESC4in1

### DISCUSSION THREAD

**Discussion by mandelyoni** [bitcraze/discussions#457] [Q&A] [answered]
Hi sir,

We're trying to connect the Bolt to a Mamba F30 mini 4in1ESC:
https://cdn.shopify.com/s/files/1/0027/2708/4144/files/3-MAMBA_Basic_F405_MINI_MK3.5.pdf?v=1610714256

We're running the propellor test, through the client, but the motor does not move.
Do we need to config in the code CONFIG_MOTORS_ESC_PROTOCOL_DSHOT ?

Please take a look in the picture below.

Thanks,  Yoni

![WhatsApp Image 2022-12-07 at 14 55 33](https://user-images.githubusercontent.com/4273822/206185436-28ed70cc-a2ef-4aa9-a7b5-84496d365235.jpeg)

**Comment by tobbeanton** [comment]
I would connect it like this:

- Wire the thick battery cables from the connector to the 4in1 ESC as this consumes most of the power
- Wire the thinner battery cables in parallel to the Bolt from the battery/ESC.
- Wire the ESC signals from the Bolt ESC connectors (or solder them) to the 4in1 on respective motor. Make sure they are mapped the same way, e.g. M1 front right motor, etc

You can test it with the propeller test but you must first set the parameter "health.propTestPWMRatio" to something like 20000 first. The OneShot125 or OneShot42 ESC protocol should most likely work with the 4in1 ESC.

**Comment by mandelyoni** [comment, marked as answer]
Hi Tobias,

Thank you for your prompt and detailed reply !
Regarding your 3rd recommendation, are you sure about it? It seems the 4in1ESC expects only the 4 commands signals (through the white connector), am I wrong? The connector is shown in the bottom of the following link: https://cdn.shopify.com/s/files/1/0027/2708/4144/files/3-MAMBA_Basic_F405_MINI_MK3.5.pdf?v=1610714256

One last thing, where can I find the propTestPWMRatio param?

Thank you a lot,  Yoni

**Reply by tobbeanton** [reply]
Yes that is what I meant, only the signal wire which is the center one.

The parameter can be found in the [param tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/parameters_tab/) in the cflient under the health category.

**Reply by mandelyoni** [reply]
> Yes that is what I meant, only the signal wire which is the center one.
> 
> The parameter can be found in the [param tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/parameters_tab/) in the cflient under the health category.

It works.  Thank you Tobias !!

**Comment by Marcox2111** [comment]
Probably the problem is that the ESC has to be armed, I had this problem too. If the protocol is BLHeli32 is not still totally clear to me how to do it in a reliability way but this:
attach a controller like the xbox one
increase the thrust with the right analog
and start to move the bolt in this way it starts to vary the thrust by itself and for some reason the esc should start.
Tell me if it is not clear, I'll try to do a video.

**Reply by tobbeanton** [reply]
I think currently we just send "zero". Maybe this is not enough for BLHeli32.
